backend/Dockerfile

```diff
@@ -6,26 +6,31 @@ EXPOSE 443
 
 FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
 
+RUN mkdir -p /var/www/.nuget/packages; chown -R www-data:www-data /var/www
+USER www-data:www-data
+
 COPY . .
 # WORKDIR /src
 # Copy the main source project files
 # COPY */*.csproj ./
 #
 # move them into the proper sub folders, based on the name of the project
 # RUN for file in $(ls *.csproj); do mkdir -p ${file%.*}/ && mv $file ${file%.*}/; done
-RUN --mount=type=cache,target=/root/.nuget/packages dotnet restore "LexBoxApi/LexBoxApi.csproj"
+RUN dotnet restore "LexBoxApi/LexBoxApi.csproj"
```

Comment:
Why did we remove the cache?

Reply:
Because its contents, created in earlier runs, were owned by root and I was trying to build as the www-data user. But that was a mistake (I was confusing the main dockerfile and the dev dockerfile), which I'm about to revert.

Reply:
Reverted by force-pushing branch back to an earlier commit, then added commit 709df14 to use usernames rather than numeric UIDs as Chris suggested. Let's just see if the naive version works after all; I thought it didn't work, but that's because I got mixed up between the dev dockerfile and the main dockerfile and made a change in the main dockerfile that really belonged in the dev one. The naive approach actually did build, but I thought it hadn't.